river/tree/mondrian/mondrian_tree_regressor.py

```diff
@@ -0,0 +1,433 @@
+                    self._update_downwards(leaf, False)
+                    return leaf
+                else:
+                    # There is no split, so we just update the node and go to the next one
+                    self._update_downwards(current_node, True)
+                    if isinstance(current_node, MondrianLeafRegressor):
+                        return current_node
+                    else:
+                        # Save the path direction to keep the tree consistent
+                        try:
+                            branch_no = current_node.branch_no(self._x)
+                            current_node = current_node.children[branch_no]
+                        except KeyError:  # Missing split feature
+                            branch_no, current_node = current_node.most_common_path()
+
+    def _go_upwards(self, leaf: MondrianLeafRegressor):
+        """Update the tree (upwards procedure).
+
+        Parameters
+        ----------
+        leaf
+            Leaf to start from when going upward.
+
+        """
+
+        current_node = leaf
+
+        if self.iteration >= 1:
+            while True:
+                current_node.update_weight_tree()
+                if current_node.parent is None:
+                    # We arrived at the root
+                    break
+                # Note that the root node is updated as well
+                # We go up to the root in the tree
+                current_node = current_node.parent
+
+    def learn_one(self, x, y):
+        # Setting current sample
+        self._x = x
+        self._y = y
+
+        # Learning step
+        leaf = self._go_downwards()
+        if self.use_aggregation:
+            self._go_upwards(leaf)
+
+        # Incrementing iteration
+        self.iteration += 1
+        return self
+
+    def predict_one(self, x):
+        """Predict the label of the samples.
+
+        Parameters
+        ----------
+        x
+            Feature vector
+        """
+
+        # If the tree hasn't seen any sample, then it should return
+        # the default empty dict
+        if not self._is_initialized:
+            return
+
+        leaf = (
+            self._root.traverse(x, until_leaf=True)
+            if isinstance(self._root, MondrianBranchRegressor)
+            else self._root
+        )
+
+        if not self.use_aggregation:
+            return self._predict(leaf)
+
+        current = leaf
+        prediction = 0.0
+
+        while True:
+            # This test is useless ?
+            if isinstance(current, MondrianLeafRegressor):
+                prediction = self._predict(current)
+            else:
+                weight = current.weight
+                log_weight_tree = current.log_weight_tree
+                w = math.exp(weight - log_weight_tree)
+                # Get the predictions of the current node
+                pred_new = self._predict(current)
+                prediction = 0.5 * w * pred_new + (1 - 0.5 * w) * prediction
```

Comment:
Does it change in the regression case?

Reply:
I don't think so. That's how it is done in the onelearn git (https://github.com/onelearn/onelearn/blob/master/onelearn/tree_methods.py)

Reply:
Ok! Thanks for the clarification :D